### PR TITLE
fix(vite): don't prepend publicPath to public assets

### LINKED ISSUE #2261: Generated image urls for files in the public directory result in 404's when running the built server.
#### Local:
------------------------------
- Operating System: `Darwin`
- Node Version:     `v17.0.1`
- Nuxt Version:     `3.0.0-27303148.ac98373`
- Package Manager:  `yarn@1.22.10`
- Bundler:          `Vite`
- User Config:      `publicRuntimeConfig`, `meta`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------
#### Remote (Slightly older version of nuxt but still has the same problem):
------------------------------
- Operating System: `Linux`
- Node Version:     `v17.0.1`
- Nuxt Version:     `3.0.0-27294839.7e5e50b`
- Package Manager:  `yarn@1.22.15`
- Bundler:          `Vite`
- User Config:      `publicRuntimeConfig`, `meta`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

When running the dev server images work fine: https://codesandbox.io/s/nuxt-3-missing-images-dev-2xj0y

However when running the `nuxi build` command and then running `(node) .output/server/index.mjs` all images result in 404s: https://codesandbox.io/s/nuxt-3-missing-images-prod-rvge4

### Describe the bug

When building the server all images urls that reference a file in the public directory get parsed and prefixed with `/_nuxt/`.

So this code (referring to the file `/project/public/images/my-logo.jpg`):
```vue
<img href="/images/my-logo.jpg" />
```
Will be turned into:
```vue
<img href="/_nuxt/images/my-logo.jpg" />
```

It seems ever since version `3.0.0-27296423.f3082ca` these urls have resulted in 404's when running the built server. (The images work fine when running the local development server as this doesn't prefix the urls)

### Additional context

Might be related to:
* #2024
* #2242
* #2135

Personally I find it a little ulgy-looking that the urls of the files I put in my public directory get prefixed with `/_nuxt/`. I would instead expect only my compiled `js` and  `css` to be prefixed like that. Although that's just personal preference to be fair.

### Logs

_No response_

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves #2261

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR sets base to be `/` or cdnURL, if set (previously `_nuxt`), and set `assetsDir` instead to `_nuxt`. This means we have the desired rewriting in the HTML for our assets: static assets aren't rewritten, but generated assets are. The only hack is that we then have to rewrite the client manifest to remove this `assetsDir` prefix (because the bundle renderer will handle re-inserting it for us).

Ideally we would have support in vite directly for this.

It produces a file structure like:
```
.output/public/_nuxt/entry-74f24f8b.mjs
.output/public/_nuxt/entry.bb100772.css
.output/public/_nuxt/logo-2.148d9522.svg
.output/public/_nuxt/manifest.json
.output/public/_nuxt/vendor-1887291f.mjs
.output/public/robots.txt
.output/public/logo.svg
```

and in the HTML:

```html
<!DOCTYPE html>
<html data-head-attrs="">

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="head:count" content="2">
  <link rel="modulepreload" href="/_nuxt/entry-74f24f8b.mjs" as="script">
  <link rel="preload" href="/_nuxt/entry.bb100772.css" as="style">
  <link rel="modulepreload" href="/_nuxt/vendor-1887291f.mjs" as="script">
  <link rel="prefetch" href="/_nuxt/logo-2.148d9522.svg">
  <link rel="stylesheet" href="/_nuxt/entry.bb100772.css">
</head>

<body data-head-attrs="">
  <div id="__nuxt">
    <div class="app" data-v-12ae6901><img src="/logo.svg" height="40" data-v-12ae6901><img
        src="/_nuxt/logo-2.148d9522.svg" height="40" data-v-12ae6901>
      <h1 class="greeting" data-v-12ae6901>Hello, <br data-v-12ae6901>Nuxt 3! </h1>
    </div>
  </div>
  <script>window.__NUXT__ = { data: {}, state: { hello: "Hello" }, _errors: {}, serverRendered: true, config: { app: { basePath: "\u002F", assetsPath: "\u002F_nuxt\u002F", cdnURL: null } } }</script>
  <script type="module" src="/_nuxt/entry-74f24f8b.mjs" defer></script>
</body>

</html>
```

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] ~~update bridge vite configuration too~~
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/vite/src/vite.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export async function bundle (nuxt: Nuxt) {
             'abort-controller': 'unenv/runtime/mock/empty'
           }
         },
-        base: nuxt.options.build.publicPath,
+        base: nuxt.options.dev ? nuxt.options.build.publicPath : nuxt.options.app.cdnURL || '/',
```
Comment (pi0) [inline packages/vite/src/vite.ts:51]: Can we please not depend on CDN URL in vite config? CDN URL should not be considered in the app build step. It is mostly a deployment configuration. (what if we always default prod base to `/` ?)

**Comment by DrewLiberty** [comment]
I am experiencing a similar problem:
public assets are not copied to correct directory OR link generate in the dom is wrong.
image
image

**Comment by danielroe** [comment]
@DrewLiberty Are you getting this behaviour with this PR?

**Comment by DrewLiberty** [comment]
> @DrewLiberty Are you getting this behaviour with this PR?

Haven't tried, how can I do it? it is not merged into main so a `npm update` won't work.

**Comment by heychazza** [comment]
> @DrewLiberty Are you getting this behaviour with this PR?

How can we try it via npm?

**Comment by pi0** [comment]
Closing in the favor of #2655

**Pull request closed without merging**